tests/unit/test_config.py

```diff
@@ -0,0 +1,170 @@
+"""Unit tests for the configuration model."""
+
+import io
+import traceback
+from typing import Type, TypeVar
+
+from yaml.parser import ParserError
+
+from ols.app.models.config import InvalidConfigurationError
+from ols.utils import config
+
+E = TypeVar("E", bound=Exception)
+
+
+def check_expected_exception(
+    yaml_stream: str, expected_exception_type: Type[E], expected_error_msg: str
+) -> None:
+    """Check that an expected exception is raised."""
+    caught_expected_exception = False
+    try:
+        config.load_config_from_stream(io.StringIO(yaml_stream))
+    except expected_exception_type as e:
+        assert str(e).startswith(expected_error_msg)
+        caught_expected_exception = True
+    except Exception:
+        print(traceback.format_exc())
+        assert False
+    assert caught_expected_exception
+
+
+def test_malformed_yaml() -> None:
+    """Check that malformed YAML is handled gracefully."""
+    check_expected_exception(
+        """[foo=123}""", ParserError, "while parsing a flow sequence"
+    )
+    check_expected_exception(
+        """foobar""", AttributeError, "'str' object has no attribute 'get'"
+    )
+    check_expected_exception(
+        """123""", AttributeError, "'int' object has no attribute 'get'"
+    )
+    check_expected_exception(
+        """12.3""", AttributeError, "'float' object has no attribute 'get'"
+    )
+    check_expected_exception(
+        """[1,2,3]""", AttributeError, "'list' object has no attribute 'get'"
+    )
+
+
+def test_invalid_config() -> None:
```

Comment:
some additional invalid config scenarios:

`xxxx_model` field(e.g. default_model) doesn't refer to a valid provider/model combination

redis configuration supplied when cache type is memory
memory config supplied when cache type is redis

we should probably also go ahead and validate that any credential_paths that are provided, can be read.  (the creds may not be valid, but at least ensure that the path the user provided exists and is readable so we can fail early if it's not)